flink-core/src/main/java/org/apache/flink/configuration/ExecutionBatchOptions.java

```diff
@@ -0,0 +1,19 @@
+package org.apache.flink.configuration;
+
+import org.apache.flink.annotation.docs.Documentation;
+
+import static org.apache.flink.configuration.ConfigOptions.key;
+
+/** Configuration options for the batch execution. */
+public class ExecutionBatchOptions {
+
+    @Documentation.Section({
+        Documentation.Sections.EXPERT_SCHEDULING,
+        Documentation.Sections.ALL_JOB_MANAGER
+    })
+    public static final ConfigOption<Boolean> ADAPTIVE_AUTO_PARALLELISM_ENABLED =
+            key("execution.batch.adaptive.auto-parallelism.enabled")
+                    .booleanType()
+                    .defaultValue(true)
+                    .withDescription("Controls whether to enable auto parallelism derivation.");
```

Comment:
->  If true, Flink will automatically decide the parallelism of operators in batch jobs.